core/main/src/state/metrics_state.rs

```diff
@@ -45,6 +51,20 @@ pub struct MetricsState {
 }
 
 impl MetricsState {
+    fn send_context_update_request(platform_state: &PlatformState) {
+        let mut extn_client = platform_state.get_client().get_extn_client().clone();
+        let metrics_context = platform_state.metrics.context.read().unwrap().clone();
+
+        if let Err(e) = extn_client
+            .request_transient(RippleContextUpdateRequest::MetricsContext(metrics_context))
+        {
+            error!(
+                "Error sending context update: RippleContextUpdateRequest::MetricsContext: {:?}",
+                e
+            );
+        }
+    }
+
```

Comment:
Not all MetricsContext fields are always immediately available (a good example might be session ID), utility function to trigger an update accordingly.